mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/KeyValueStore.java

```diff
@@ -214,4 +224,61 @@ public boolean deleteAll(String tableName, String partitionKey) {
             return false;
         }
     }
+
+    class InMemoryStore implements KeyValueStore {
+
+        // table -> partitionKey -> secondaryKey -> data
+        private final Map<String, Map<String, Map<String, String>>> store = new ConcurrentHashMap<>();
+
+        @Override
+        public List<String> getAllPartitionKeys(String tableName) throws IOException {
+            try {
+                return store.get(tableName).keySet().stream().collect(Collectors.toList());
+            } catch (Exception e) {
+                throw new IOException(e);
+            }
+        }
+
+        @Override
+        public Map<String, String> getAll(String tableName, String partitionKey)
+            throws IOException {
+            if (store.get(tableName) == null) {
+                return Collections.emptyMap();
+            } else if (store.get(tableName).get(partitionKey) == null) {
+                return Collections.emptyMap();
+            } else {
+                return store.get(tableName).get(partitionKey);
+            }
+        }
+
+        @Override
+        public boolean upsertAll(String tableName, String partitionKey, Map<String, String> all,
+            Duration ttl) throws IOException {
+            store.putIfAbsent(tableName, new ConcurrentHashMap<>());
+            store.get(tableName).put(partitionKey, new ConcurrentHashMap<>(all));
+            return true;
+        }
+
+        @Override
+        public boolean delete(String tableName, String partitionKey, String secondaryKey)
+            throws IOException {
+            if (store.containsKey(tableName) && // table exists
+                store.get(tableName).containsKey(partitionKey) && // partitionKey exists
+                store.get(tableName).get(partitionKey).containsKey(secondaryKey)) { // secondaryKey exists
+                store.get(tableName).get(partitionKey).remove(secondaryKey);
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public boolean deleteAll(String tableName, String partitionKey) throws IOException {
+            if (store.containsKey(tableName) && // table exists
+                store.get(tableName).containsKey(partitionKey)) { // partitionKey exists
+                store.get(tableName).remove(partitionKey);
+                return true;
```

Comment:
@hmit Can you check if the semantics of this is consistent with the KeyValueStore assumptions